src/unlimited.cpp

```diff
@@ -1212,23 +1213,27 @@ bool CheckAndRequestExpeditedBlocks(CNode* pfrom)
     {
       BOOST_FOREACH(string& strAddr, mapMultiArgs["-expeditedblock"]) 
         {
-          // Add the peer's listening port if it is empty
-          int pos1 = strAddr.rfind(":");
-          int pos2 = strAddr.rfind("]:");
-          if (pos1 <= 0 && pos2 <= 0)
-              strAddr += ':' + boost::lexical_cast<std::string>(pfrom->addrFromPort);
-
           string strListeningPeerIP;
           string strPeerIP = pfrom->addr.ToString();
-          pos1 = strPeerIP.rfind(":");
-          pos2 = strPeerIP.rfind("]:");
-          // Handle both ipv4 and ipv6 cases
-          if (pos1 <= 0 && pos2 <= 0) 
-              strListeningPeerIP = strPeerIP + ':' + boost::lexical_cast<std::string>(pfrom->addrFromPort);
-          else if (pos1 > 0)
-              strListeningPeerIP = strPeerIP.substr(0, pos1) + ':' + boost::lexical_cast<std::string>(pfrom->addrFromPort);
-          else
-              strListeningPeerIP = strPeerIP.substr(0, pos2) + ':' + boost::lexical_cast<std::string>(pfrom->addrFromPort);
+          // Add the peer's listening port if it was provided (only misbehaving clients do not provide it)
+          if (pfrom->addrFromPort != 0)
+            {
+              int pos1 = strAddr.rfind(":");
+              int pos2 = strAddr.rfind("]:");
+              if (pos1 <= 0 && pos2 <= 0)
+                strAddr += ':' + boost::lexical_cast<std::string>(pfrom->addrFromPort);
```

Comment:
Guys as some of you have mentioned, it does not make sense to fix the formatting at the same time as logic changes.
All of these lines are showing up changed because they were indented since they are now inside an if() block that verifies that the client has provided his port in the pfrom->addrFromPort by sending a BUVERSION message before we request expedited blocks from it.  The material change here is to not use addrFromPort if it has not yet been initialized.